internal/commands/app/create_inputs_test.go

```diff
@@ -0,0 +1,272 @@
+		Name:        "test-app",
+	}
+
+	for _, tc := range []struct {
+		description    string
+		inputs         createInputs
+		from           from
+		procedure      func(c *expect.Console)
+		findAppErr     error
+		expectedName   string
+		expectedFilter realm.AppFilter
+		expectedErr    error
+	}{
+		{
+			description:  "should return name if name is set",
+			inputs:       createInputs{newAppInputs: newAppInputs{Name: testApp.Name}},
+			procedure:    func(c *expect.Console) {},
+			expectedName: testApp.Name,
+		},
+		{
+			description:    "should use from app for name if name is not set",
+			from:           from{testApp.GroupID, testApp.ID},
+			procedure:      func(c *expect.Console) {},
+			expectedName:   testApp.Name,
+			expectedFilter: realm.AppFilter{GroupID: testApp.GroupID, App: testApp.ID},
+		},
+		{
+			description:    "should error when finding app",
+			from:           from{testApp.GroupID, testApp.ID},
+			procedure:      func(c *expect.Console) {},
+			findAppErr:     errors.New("realm client error"),
+			expectedFilter: realm.AppFilter{GroupID: testApp.GroupID, App: testApp.ID},
+			expectedErr:    errors.New("realm client error"),
+		},
+	} {
+		t.Run(tc.description, func(t *testing.T) {
+			var appFilter realm.AppFilter
+			rc := mock.RealmClient{}
+			rc.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+				appFilter = filter
+				return []realm.App{testApp}, tc.findAppErr
+			}
+
+			_, console, _, ui, consoleErr := mock.NewVT10XConsole()
+			assert.Nil(t, consoleErr)
+			defer console.Close()
+
+			doneCh := make(chan (struct{}))
+			go func() {
+				defer close(doneCh)
+				tc.procedure(console)
+			}()
+
+			name, err := tc.inputs.resolveAppName(ui, rc, tc.from)
+
+			console.Tty().Close() // flush the writers
+			<-doneCh              // wait for procedure to complete
+
+			assert.Equal(t, tc.expectedErr, err)
+			assert.Equal(t, tc.expectedName, name)
+			assert.Equal(t, tc.expectedFilter, appFilter)
+		})
+	}
+}
+
+func TestAppCreateInputsResolveProject(t *testing.T) {
+	testApp := realm.App{
+		ID:          primitive.NewObjectID().Hex(),
+		GroupID:     primitive.NewObjectID().Hex(),
+		ClientAppID: "test-app-abcde",
+		Name:        "test-app",
+	}
+
+	for _, tc := range []struct {
+		description     string
+		inputs          createInputs
+		procedure       func(c *expect.Console)
+		groupsErr       error
+		expectedProject string
+		expectedErr     error
+	}{
+		{
+			description:     "should return project if project is set",
+			inputs:          createInputs{newAppInputs: newAppInputs{Project: testApp.GroupID}},
+			procedure:       func(c *expect.Console) {},
+			expectedProject: testApp.GroupID,
+		},
+		{
+			description: "should prompt for project if project is not set",
+			procedure: func(c *expect.Console) {
+				c.ExpectString("Atlas Project")
+				c.Send(testApp.GroupID)
+				c.SendLine(" ")
+				c.ExpectEOF()
+			},
+			expectedProject: testApp.GroupID,
+		},
+		{
```

Comment:
same thing about loaded test case `struct`s and not needing pseudo-terminals for tests where we aren't prompting for anything.  these three tests all seem easy enough to test in their own `t.Run` blocks